### PR TITLE
Forupstream

### PR DESCRIPTION
**Pull request by fzago-cray**

### CODE REVIEW

File: lib/scanner.c [hunk 1]
```diff
@@ -9,7 +9,7 @@
 #define FLEX_SCANNER
 #define YY_FLEX_MAJOR_VERSION 2
 #define YY_FLEX_MINOR_VERSION 5
-#define YY_FLEX_SUBMINOR_VERSION 39
+#define YY_FLEX_SUBMINOR_VERSION 35
```
Comment (hyperrealm) [inline lib/scanner.c]: Any reason for moving back to an older version of flex? I'd prefer not to do that.

File: tests/tests.c [hunk 2]
```diff
@@ -207,14 +207,274 @@ TT_TEST(ParseInvalidStrings)
...
+	FILE *f;
+	config_t cfg;
+	int rc;
+	int ival;
+	long long llval;
+
+	/* int = 5 */
+	buf = "someint=5;";
+
+	f = fmemopen(buf, strlen(buf), "r");
```
Comment (hyperrealm) [inline tests/tests.c]: I am not sure how portable fmemopen() is. Why not just use config_read_string() below?

File: tests/tests.c [hunk 3]
```diff
@@ -207,14 +207,274 @@ TT_TEST(ParseInvalidStrings)
 
 /* ------------------------------------------------------------------------- */
 
+TT_TEST(BigInt1)
+{
+	char *buf;
```
Comment (hyperrealm) [inline tests/tests.c]: Please use 2-space indent in this file.

File: lib/scanner.c (hunk 1 above)
Comment (fzago-cray) [inline lib/scanner.c]: No reason besides that it's what I have installed. If you accept this patch, you could regenerate that file with your version before commiting.

File: tests/tests.c (hunk 3 above)
Comment (fzago-cray) [inline tests/tests.c]: Sure.

File: tests/tests.c (hunk 2 above)
Comment (fzago-cray) [inline tests/tests.c]: Yes, that looks better. I wasn't aware of that function.

**Comment by fzago-cray** [comment]
All the comment have been addressed, and the pull request has been updated (I think).

**Comment by hyperrealm** [comment]
OK, I'll try to get this done over the weekend. Thanks for this nice patch!

**Pull request merged**